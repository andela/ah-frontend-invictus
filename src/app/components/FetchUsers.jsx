import React from 'react';
import PropTypes from 'prop-types';
import Navbar from './Navbar';

const FetchUsers = (props) => {
  const isFollowing = (obj) => {
    let list = props.state.following.profiles;
    var i;
    for (i = 0; i < list.length; i++) {
      if (list[i].user === obj.user) {
        return true;
      }
    } return false;
  };
  const allUsers = props.state.profiles.profiles.map(profile => (<p key={profile.user} className="card-article-link" to={`/users/${profile.user}`}>
    <div className="card-article" style={{ width: 21.85 + 'rem' }}><div className="card-body">
      <div><p className="card-publish-date"><b>Updated At: </b>{profile.updated_at.substring(0, 10)}</p></div> <p className="card-text"><b>Bio:</b> {profile.bio}</p><p className="card-footer-text"><b>FirstName:</b> {profile.firstname}</p>
      <p className="card-footer-text"><b>LastName:</b> {profile.lastname}</p><p className="card-footer-text"><b>UserName:</b> {profile.user}</p>
      <div className="card-body-footer">
        {isFollowing(profile) ? (<a href="#"><button type="button" className="btn btn-outline-success" onClick={() => { let username = profile.user; props.state.unfollowUsers(username); }}>UnFollow</button></a>) : (<a href="#"><button type="button" className="btn btn-outline-success"
          onClick={() => { let username = profile.user; props.state.followUsers(username); }}>Follow</button></a>)}<div className="all-tags"><hr />
        </div></div></div></div>
  </p>
  ));
  return (
    <div>
      <Navbar />
      <div className="container">
        <h2 className="articles-heading">All User Profiles</h2><hr />
        <div className="row">{allUsers}</div>
      </div>
    </div>
  );
};

FetchUsers.propTypes = {
  unfollowUsers: PropTypes.func,
  followUsers: PropTypes.func
};
export default FetchUsers;

