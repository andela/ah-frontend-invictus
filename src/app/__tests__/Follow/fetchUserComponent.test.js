import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import FetchUsers from '../../components/FetchUsers';

const props = {
  state: {
    "profiles": {
      "profiles": [{
        "id": 1,
        "bio": "",
        "firstname": "",
        "lastname": "",
        "updated_at": "2019-06-17T02:05:35.588872Z",
        "user": "sanygreen"
      }]
    },
    "following": {
      "profiles": [{
        "id": 1,
        "bio": "",
        "firstname": "",
        "lastname": "",
        "updated_at": "2019-06-17T02:05:35.588872Z",
        "user": "sanygreen"
      }]
    }
  }
};
describe('<FetchUsers />', () => {
  let wrapper;
  it('renders the user profile page', () => {
    wrapper = shallow(<FetchUsers {...props} />);
  });

  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
