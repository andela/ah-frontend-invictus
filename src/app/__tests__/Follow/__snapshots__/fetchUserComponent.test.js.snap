// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FetchUsers /> matches snapshot 1`] = `
<div>
  <Connect(NavbarComponent) />
  <div
    className="container"
  >
    <h2
      className="articles-heading"
    >
      All User Profiles
    </h2>
    <hr />
    <div
      className="row"
    >
      <p
        className="card-article-link"
        key="sanygreen"
        to="/users/sanygreen"
      >
        <div
          className="card-article"
          style={
            Object {
              "width": "21.85rem",
            }
          }
        >
          <div
            className="card-body"
          >
            <div>
              <p
                className="card-publish-date"
              >
                <b>
                  Updated At: 
                </b>
                2019-06-17
              </p>
            </div>
             
            <p
              className="card-text"
            >
              <b>
                Bio:
              </b>
               
            </p>
            <p
              className="card-footer-text"
            >
              <b>
                FirstName:
              </b>
               
            </p>
            <p
              className="card-footer-text"
            >
              <b>
                LastName:
              </b>
               
            </p>
            <p
              className="card-footer-text"
            >
              <b>
                UserName:
              </b>
               
              sanygreen
            </p>
            <div
              className="card-body-footer"
            >
              <a
                href="#"
              >
                <button
                  className="btn btn-outline-success"
                  onClick={[Function]}
                  type="button"
                >
                  UnFollow
                </button>
              </a>
              <div
                className="all-tags"
              >
                <hr />
              </div>
            </div>
          </div>
        </div>
      </p>
    </div>
  </div>
</div>
`;
